src/main/java/com/yelp/nrtsearch/server/luceneserver/index/handlers/FieldUpdateHandler.java

```diff
@@ -0,0 +1,201 @@
+/*
+ * Copyright 2022 Yelp Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.yelp.nrtsearch.server.luceneserver.index.handlers;
+
+import com.yelp.nrtsearch.server.grpc.Field;
+import com.yelp.nrtsearch.server.grpc.FieldDefRequest;
+import com.yelp.nrtsearch.server.grpc.FieldDefResponse;
+import com.yelp.nrtsearch.server.grpc.FieldType;
+import com.yelp.nrtsearch.server.luceneserver.IndexState;
+import com.yelp.nrtsearch.server.luceneserver.field.FieldDef;
+import com.yelp.nrtsearch.server.luceneserver.field.FieldDefCreator;
+import com.yelp.nrtsearch.server.luceneserver.field.IndexableFieldDef;
+import com.yelp.nrtsearch.server.luceneserver.field.VirtualFieldDef;
+import com.yelp.nrtsearch.server.luceneserver.index.FieldAndFacetState;
+import com.yelp.nrtsearch.server.luceneserver.index.IndexStateManager;
+import com.yelp.nrtsearch.server.luceneserver.script.ScoreScript;
+import com.yelp.nrtsearch.server.luceneserver.script.ScriptService;
+import com.yelp.nrtsearch.server.luceneserver.script.js.JsScriptEngine;
+import com.yelp.nrtsearch.server.utils.ScriptParamsUtils;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import org.apache.lucene.search.DoubleValuesSource;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/** Static helper class to handle a request to add/update fields. */
+public class FieldUpdateHandler {
+  private static final Logger logger = LoggerFactory.getLogger(FieldUpdateHandler.class);
+
+  private FieldUpdateHandler() {}
+
+  /**
+   * Handle a FieldDefRequest.
+   *
+   * @param indexStateManager state manager for index
+   * @param request request message
+   * @return response message
+   * @throws IOException on error committing state
+   */
+  public static FieldDefResponse handle(
```

Comment:
Both the register and update fields endpoints use this handler. This is also the case for the legacy state management, since `UpdateFieldsHandler` defers to `RegisterFieldsHandler`.

Hopefully we can get some kind of update support added here soon.